File: JavaBase/이분탐색/LowerBound.java
public class LowerBound {
    public static void main(String[] args) {
        int[] nums = { 1, 2, 3, 4, 4, 4, 5, 6, 7, 8 };
        int start = 0;
        int end = nums.length - 1;
        int M = 4;
        while (start <= end) {
            int mid = (start + end) / 2;
            int temp = nums[mid];
            if (temp >= M) {
                end = mid - 1;
            } else {
                start = mid + 1;
            }
        }
        System.out.print(end + 1);
    }
}

File: 11/11102c/lower_bound.cpp
#include <iostream>
using namespace std;
int lower_bound(int *arr, int N, int target)
{
    int start = 0;
    int end = N;
    int mid;
    while (end > start)
    {
        mid = (start + end) / 2;
        if (arr[mid] < target)
            start = mid + 1;
        else
            end = mid;
    }
    return end;
}
int main()
{
    //             0  1  2  3  4  5  6  7  8  9
    int arr[10] = {1, 2, 4, 5, 6, 6, 7, 7, 7, 9};
    cout << lower_bound(arr, 10, 7);
    return 0;
}
